src/tests/unit/components/slider-with-input/SliderWithInput.spec.jsx

```diff
@@ -0,0 +1,80 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { afterEach, beforeEach, expect, it, vi } from 'vitest'
+import SliderWithInput from '~/components/slider-with-input/SliderWithInput'
+
+const title = 'Price'
+const defaultValue = 8
+const min = 0
+const max = 100
+const onChangeSlider = vi.fn()
+
+describe('SliderWithInput component', () => {
+  beforeEach(() => {
+    render(
+      <SliderWithInput
+        defaultValue={defaultValue}
+        max={max}
+        min={min}
+        onChange={onChangeSlider}
+        title={title}
+      />
+    )
+  })
+
+  afterEach(() => vi.clearAllMocks())
+
+  it('should render correctly', () => {
+    const titleText = screen.getByText(title)
+    const slider = screen.getByRole('slider')
+    const input = screen.getByRole('textbox')
+
+    expect(titleText).toBeInTheDocument()
+    expect(slider).toBeInTheDocument()
+    expect(input).toBeInTheDocument()
+  })
+
+  it('should call onChange when slider is moved', async () => {
+    const slider = screen.getByRole('slider')
+    expect(slider).toHaveValue('8')
+
+    await userEvent.type(slider, '50')
+    expect(slider).toHaveValue('50')
+
+    await new Promise((r) => setTimeout(r, 600))
```

Comment:
use this syntax to wait for your assertion
await waitFor(() => {    
    expect(container.queryElement('span')).toBeDefined(); // your assertion here
  })

Reply:
I updated the syntax with waitFor, could you please check?